src/memex/storage.py

```diff
@@ -173,12 +173,12 @@ def update_annotation(session, id_, data):
     if document:
         document_uri_dicts = document['document_uri_dicts']
         document_meta_dicts = document['document_meta_dicts']
-        models.update_document_metadata(
-            session,
-            annotation.target_uri,
-            document_meta_dicts,
-            document_uri_dicts,
-            updated=updated)
+        document = models.update_document_metadata(session,
+                                                   annotation.target_uri,
+                                                   document_meta_dicts,
+                                                   document_uri_dicts,
+                                                   updated=updated)
+        annotation.document_id = document.id
```

Comment:
As in the previous commit, maybe add a `FIXME` here to come back and update this to just do `annotation.document = document` once we've updated the relationship property.